slashing/program/src/processor.rs

```diff
@@ -13,21 +13,28 @@ use {
     serde::Deserialize,
     solana_program::{
         account_info::{next_account_info, AccountInfo},
-        clock::Slot,
+        clock::{Slot, DEFAULT_SLOTS_PER_EPOCH},
         entrypoint::ProgramResult,
         msg,
         program_error::ProgramError,
         pubkey::Pubkey,
+        sysvar::{clock::Clock, Sysvar},
     },
 };
 
 fn verify_proof_data<'a, T>(slot: Slot, pubkey: &Pubkey, proof_data: &'a [u8]) -> ProgramResult
 where
     T: SlashingProofData + Deserialize<'a>,
 {
-    if proof_data.len() < T::PROOF_TYPE.proof_account_length() {
-        return Err(ProgramError::InvalidAccountData);
+    // Statue of limitations is 1 epoch
+    let clock = Clock::get()?;
+    let Some(elapsed) = clock.slot.checked_sub(slot) else {
+        return Err(ProgramError::ArithmeticOverflow);
+    };
+    if elapsed > DEFAULT_SLOTS_PER_EPOCH {
```

Comment:
Can you use the leader schedule sysvar instead of hard-coding the slot-length? In case we ever want to change this, we'll be screwed. You can just `EpochSchedule::get()`

Reply:
Good call 09235dac6e62961e673d281fc36063514550116e